### PR TITLE
Upgrade to the latest fc-go-sdk

### PR DESCRIPTION
**Pull request by henry118**
Signed-off-by: Henry Wang <henwang@amazon.com>

*Description of changes:*

Update `firectl` to the latest `firecracker-go-sdk`, which is based on fc 1.0 release

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by kzys** [comment]
Let me fix the test first. It has been broken since I updated Go on the host.

File: options.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (opts *options) getFirecrackerConfig() (firecracker.Config, error) {
 		MachineCfg: models.MachineConfiguration{
 			VcpuCount:   firecracker.Int64(opts.FcCPUCount),
 			CPUTemplate: models.CPUTemplate(opts.FcCPUTemplate),
-			HtEnabled:   firecracker.Bool(htEnabled),
+			Smt:         firecracker.Bool(htEnabled),
```
Comment (kzys) [inline options.go:157]: Let's rename `htEnabled` as well.

Comment (henry118) [reply options.go]: Updated

File: options.go [hunk 2]
```diff
@@ -48,7 +48,7 @@ type options struct {
 	FcLogFifo          string   `long:"vmm-log-fifo" description:"FIFO for firecracker logs"`
 	FcLogLevel         string   `long:"log-level" description:"vmm log level" default:"Debug"`
 	FcMetricsFifo      string   `long:"metrics-fifo" description:"FIFO for firecracker metrics"`
-	FcDisableHt        bool     `long:"disable-hyperthreading" short:"t" description:"Disable CPU Hyperthreading"`
+	FcDisableSmt       bool     `long:"disable-smt" short:"t" description:"Disable CPU Simultaneously Multi Threading"`
```
Comment (Kern--) [inline options.go:51]: nit: Multithreading

**Review by Kern-- — APPROVED** [review]

File: options.go (hunk 2 above)
Comment (henry118) [reply options.go]: updated.

**Review by kzys — APPROVED** [review]

**Pull request merged**